source/blog/2019-03-02-update-on-module-unification-and-octane.md

```diff
@@ -0,0 +1,496 @@
+
+Feedback from the community was that pods felt more productive than the
+classic layout. However, pods had several problems that needed to be
+addressed before it could be enabled by default.
+
+The effort to address the design flaws of pods led to the [Module Unification
+RFC][mu], a ground-up rethink of how the file system should work in Ember
+apps. Importantly, this design grappled with overhauling Ember's resolver and
+dependency injection systems as well, which was just as important as
+shuffling around where particular files went on disk.
+
+[mu]: https://emberjs.github.io/rfcs/0143-module-unification.html
+
+The MU RFC was merged at the end of 2016, but new Ember apps are still
+generated with the classic layout. So, why can't we flip the switch on
+enabling Module Unification-style layout by default today?
+
+Remember that MU described not just a new file system but a significant
+overhaul of Ember's resolver. Implementing MU was a **huge** initiative that,
+while often hidden, touched nearly every part of the framework. Despite this
+large scope, and thanks to the incredible amount of time and energy our
+community devoted to the work, MU implementation has made great progress, and
+nearly everything needed to make Module Unification work has landed in Ember.
+
+**Nearly** everything. When we merged the Roadmap RFC last year, there was one
+last major piece of MU that still needed to be designed: component
+namespacing, or how to refer to components that come from other addons in
+templates.
+
+While we had a plan, a shift in the JavaScript ecosystem sent us back to the
+drawing board. And the harder we worked on solving this last piece, the more
+we realized that fundamental aspects of the overall design needed to be
+rethought.
+
+### Namespaces and Scoped Packages
+
+Besides files being spread out, another problem with the classic layout is
+that everything goes into one big global pool of names.
+
+So, for example, Ember Data defines a service called `store` that you can
+inject into components and services. If you install another addon that also
+has a service called `store`, there's no easy way to use both.
+
+Similarly, if your app has a component called `small-button` and you install
+an addon that also has a component called `small-button`, you have no way to
+tell Ember which one you mean when you type `{{small-button}}` in a template.
+
+One of the key benefits of the MU design is that names are no longer global,
+but namespaced to the package where they come from. So, for example, if an
+npm package called `ember-ui-library` contains a component called
+`small-button`, the MU RFC proposed referring to it in your template as
+`{{ember-ui-library::small-button}}`.
+
+This worked pretty well, with the downside that it could sometimes be
+verbose, particularly when components were invoked with a block:
+
+```handlebars
+{{#ember-ui-library::small-button}}
+  ...
+{{/ember-ui-library::small-button}}
+```
```

Comment:
I would _probably_ just skip this, since scoped packages is a much bigger problem, and this draws some attention to "doesn't nested components above have the same issue?". (IMO, not really, but still.)